### DISCUSSION TITLE
FastDDS 3 Compatibility with ROS2

### DISCUSSION THREAD

**Discussion by bshantam97** [eProsima/Fast-DDS#5259] [Q&A] [answered]
Hello, I recently Upgraded to FastDDS 3 and was wondering whether ROS2 Can be built with FastDDS3 as a target considering the namespaces have been changed. In additionis there an rmw available for fastdds 3 compatible with ros2?

**Comment by JesusPoderoso** [comment, marked as answer]
Hi @bshantam97, the required changes are work in progress (almost finished).  You can keep track of them in the following PR. 
- https://github.com/ros2/rmw_fastrtps/pull/776

You may be interested in [Vulcanexus](https://vulcanexus.org/), a ROS 2 all-in-one tool set that allows users to build robotics applications combining the unique [Vulcanexus elements](https://vulcanexus.org/product) with the ROS 2 libraries, having the latest Fast DDS release as its fixed middleware implementation. 
A new _Vulcanexus Jazzy_ version with _Fast DDS v3_ will be released soon.
Contact info@eprosima.com for more information.

**Reply by bshantam97** [reply]
Thank you @JesusPoderoso!

**Reply by bshantam97** [reply]
@JesusPoderoso Hello, when is the estimated time when FastDDS rmw will be merged into ROS2 ? Thanks

**Reply by JesusPoderoso** [reply]
Hi @bshantam97. Unfortunately, that procedure is not on our side and we cannot estimate the revision and merge time.
You may be interested in [Vulcanexus](https://vulcanexus.org/) in the meantime.
